TACC/Core-Portal
# Task/FP-387: UI Fixes

## Overview: ##

Fix a bunch of UI issues with FP-387.

See [request on PR for FP-387](https://github.com/TACC/Frontera-Portal/pull/29#pullrequestreview-420205889).

## PR Status: ##

* Work in Progress.
* Hold.

## Related Jira tickets: ##

* [FP-387](https://jira.tacc.utexas.edu/browse/FP-387)

## Summary of Changes: ##

Generic
- Add comments for relevant CSS I previously added.

`client/src/components/Jobs/Jobs.js`
- Change text from "Job ID" to "Job Name".
- Add class names so CSS can truncate specific text.

`client/src/components/Jobs/Jobs.scss`
- Simplify overly-specific and inaccurate selector (remove `.InfiniteScrollTable` from selector).
    - "overly-specific", because the selector matches without the extra class
    - "inaccurate", because the column widths are for the jobs-view, regardless what kind of table it uses
- Update column width styles to match new columns.
- Use new utility class.

`client/src/components/_common/InfiniteScrollTable/InfiniteScrollTable.scss`
- Remove padding that seemed to be unnecessary.
- Automatically expand expected inline elements for `.InfiniteScrollTable` cells.
- Make all rows of `o-fixed-header-table` (thus `.InfiniteScrollTable`) use CSS Flexbox.
- Support expansion of elements within `.o-fixed-header-table` cells.

`client/src/styles/README.md`
- Create and document styles/ directory.

`client/src/styles/trumps/_u-ellipsis.scss`
- Create new utility class `u-ellipsis`.
    - used by `Jobs.scss`
    - expected to be imported for manual instances of similar code

## Testing Steps: ##

1. Load Dashboard.
1. (#41, #45) Ensure "Job Name" and "Output Location" values are truncated:
    - before overflowing cell space
1. (#42) Ensure "Job Name" is the first column header text of "My Recent Jobs" table.
1. (#44) Ensure column headers and cell content all fit in the Jobs table.

## Notes: ##

### Tasks

- [x] [temp. width change for jobs (pending JS solution)](https://github.com/TACC/Frontera-Portal/pull/45#issuecomment-637109633)
- [x] [fix non-data row formatting](https://github.com/TACC/Frontera-Portal/pull/45#issuecomment-637109872)

### Extra Fixes
* bug with `noDataText` `PropType` validation

### Screenshots
- Data\
  ![FP-387 Data](https://user-images.githubusercontent.com/62723358/83930630-35d2e700-a75e-11ea-8e07-7d8a0a8d4da4.png)
- Empty\
  ![FP-387 Empty](https://user-images.githubusercontent.com/62723358/83930632-37041400-a75e-11ea-8dca-361e26bf2b83.png)
- Loading\
  ![FP-387 Load](https://user-images.githubusercontent.com/62723358/83930633-37041400-a75e-11ea-9979-798f9ec524be.png)\
  (smaller icon now, but matches allowed sizes; see [UI - Components - Icon](https://confluence.tacc.utexas.edu/x/SQCzBw))

### Known Issues
- icon alignment ([FP-455](https://jira.tacc.utexas.edu/browse/FP-455))
- truncation is not ideal (JavaScript solution will be available after [FP-425](https://jira.tacc.utexas.edu/browse/FP-425))
